gateway/ln-gateway/src/lib.rs

```diff
@@ -163,40 +172,53 @@ where
 
 pub struct LnGateway {
     config: GatewayConfig,
-    actors: HashMap<FederationId, Arc<GatewayActor>>,
+    actors: Mutex<HashMap<FederationId, Arc<GatewayActor>>>,
     ln_client: Arc<dyn LnRpc>,
     webserver: tokio::task::JoinHandle<axum::response::Result<()>>,
     receiver: mpsc::Receiver<GatewayRequest>,
     client_builder: GatewayClientBuilder,
+    // TODO: consider wrapping bind_addr, and node_pubkey in GatewayConfig
+    bind_addr: SocketAddr,
+    pub_key: PublicKey,
 }
 
 impl LnGateway {
     pub fn new(
         config: GatewayConfig,
         ln_client: Arc<dyn LnRpc>,
+        client_builder: GatewayClientBuilder,
+        // TODO: consider encapsulating message channel within LnGateway
         sender: mpsc::Sender<GatewayRequest>,
         receiver: mpsc::Receiver<GatewayRequest>,
+        // TODO: consider wrapping bind_addr, and node_pubkey in GatewayConfig
         bind_addr: SocketAddr,
-        client_builder: GatewayClientBuilder,
+        pub_key: PublicKey,
     ) -> Self {
         // Run webserver asynchronously in tokio
         let webserver = tokio::spawn(run_webserver(config.password.clone(), bind_addr, sender));
 
         Self {
             config,
-            actors: HashMap::new(),
+            actors: Mutex::new(HashMap::new()),
             ln_client,
             webserver,
             receiver,
             client_builder,
+            bind_addr,
+            pub_key,
         }
     }
 
     fn select_actor(&self, federation_id: FederationId) -> Result<Arc<GatewayActor>> {
-        self.actors
-            .get(&federation_id)
-            .cloned()
-            .ok_or(LnGatewayError::UnknownFederation)
+        if let Ok(actors) = self.actors.lock() {
```

Comment:
Wondering how non-async mutexes will play with async code. My intuition is we should avoid mixing the both and just go with `tokio` mutexes. Looking for secondary opinions here @dpc @jkitman

Reply:
I'm interpreting this t[okio guide](https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html) and it seems sync mutexes might be reasonably applicable in this async context.

- the `actors` mutex operates purely on data. It's not a lock on IO resources
- gateway `select_actor` applied by most gateway rpcs needs to be a fast lock on the actors mutes, but I'm not use I need to keep the mutex across the rpc await. I expect these to be high frequency RPCs
- `/register_federation` places the most important guard on the mutex! When registering a new federation, `select_actor` is nondeterministic. But this is a rare RPC

If the sync mutex turns out to be performant and with no deadlocks, should we keep it instead of async mutex which this tokio guide cautions is more expensive?